python/az/aro/azext_aro/custom.py

```diff
@@ -162,14 +178,21 @@ def aro_delete(cmd, client, resource_group_name, resource_name, no_wait=False):
         # delete the cluster. Where possible, fix this before attempting
         # deletion.
         if rp_client_sp:
-            sp_id = rp_client_sp.object_id
-            assign_contributor_to_vnet(cmd.cli_ctx, vnet, sp_id)
-            assign_contributor_to_routetable(cmd.cli_ctx,
-                                             worker_subnets | {master_subnet},
-                                             sp_id)
-    except (CloudError, HttpOperationError) as e:
-        # Default to old deletion behaviour in case operations throw an
-        # exception above. Log the error.
+            for resource in resources:
+                # Create role assignments if they don't exist
+                # Note: This logic assumes that if we cannot if the role assignment
+                # we assume that it already exists
+                resource_contributor_exists = True
+            
+                try:
+                    resource_contributor_exists = has_contributor_on_resource(cmd.cli_ctx, resource, rp_client_sp.object_id)
+                except CloudError:
+                    pass
+
+                if not resource_contributor_exists:
+                    assign_contributor_to_resource(cmd.cli_ctx, resource, rp_client_sp.object_id)
```

Comment:
Instead of putting everything in a list, I separated out the functions into two main parts.  The first `try/except` tries to uncover information about the cluster itself.  If that fails, it will move to the second portion.  The second `try/except` tries to assign permission and update the cluster, but if it fails for some reason will error out and not perform any action.  

Thought this might be a little less repetitive.  Let me know your thoughts - can take another pass at it :)